### PR TITLE
o1vm/riscv32: implement mul_hi_signed

### PR DESCRIPTION
**Pull request by dannywillems**

### CODE REVIEW

File: o1vm/src/interpreters/riscv32im/witness.rs [hunk 1]
```diff
@@ -441,6 +441,21 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
...
+        x: &Self::Variable,
+        y: &Self::Variable,
+        position: Self::Position,
+    ) -> Self::Variable {
+        let x: u32 = (*x).try_into().unwrap();
+        let y: u32 = (*y).try_into().unwrap();
+        let res = (((x as i32) as i64) * ((y as i32) as i64)) as u64;
+        let res = (res >> 32) as u32;
+        let res = res as u64;
+        self.write_column(position, res);
```
Comment (svv232) [inline o1vm/src/interpreters/riscv32im/witness.rs:455]: why not 

```suggestion
        let x: i32 = (*x).try_into().unwrap();
        let y: i32 = (*y).try_into().unwrap();
        let res = (x as i64) * (y as i64);
        let res = (res >> 32) as i32;
        let res = res as u64;
        self.write_column(position, res);
```

Comment (svv232) [reply o1vm/src/interpreters/riscv32im/witness.rs:455]: i'm just preserving the signed bit for as long as possible through each operation.

Comment (dannywillems) [reply o1vm/src/interpreters/riscv32im/witness.rs]: I updated.

Comment (dannywillems) [reply o1vm/src/interpreters/riscv32im/witness.rs]: See 75503032b4306e3743f936c413c3c4e4857abced

**Review by svv232 — APPROVED** [review]

**Pull request merged**